nupic/embodied/agents/curious_ppo_agent.py

```diff
@@ -356,6 +333,49 @@ def update(self):
                 # Log buffer video from first env
                 info["observations"] = wandb.Video(sample_video, fps=12, format="gif")
 
+        return info
+
+    def collect_rewards(self, normalize=False):
+        if normalize:
+            discounted_rewards = np.array(
+                [
+                    self.reward_forward_filter.update(rew)
+                    for rew in self.rollout.buf_rewards.T
+                ]
+            )
+            # rewards_mean, rewards_std, rewards_count
+            rewards_mean, rewards_std, rewards_count = mpi_moments(
+                discounted_rewards.ravel()
+            )
+            # reward forward filter running mean std
+            self.reward_stats.update_from_moments(
+                rewards_mean, rewards_std ** 2, rewards_count
+            )
+            return self.rollout.buf_rewards / np.sqrt(self.reward_stats.var)
+        else:
+            # Copy directly from buff rewards
+            return np.copy(self.rollout.buf_rewards)
+
+
+    def update(self):
+        """Calculate losses and update parameters based on current rollout.
+
+        Returns
+        -------
+        info
+            Dictionary of infos about the current update and training statistics.
+
+        """
+        rews = self.collect_rewards(normalize=self.normrew)
+
+        # Calculate advantages using the current rewards and value estimates
+        self.calculate_advantages(
+            rews=rews, use_done=self.use_done, gamma=self.gamma, lam=self.lam
+        )
+
+        # TODO: we are logging buf_advantages before normalizing, is that correct?
```

Comment:
We could log this additionally but I think it won't be very informative to log the mean and std after normalization since it should always be 0 and 1 respectively.